code/modules/mob/living/carbon/carbon.dm

```diff
@@ -435,6 +435,12 @@
 	else
 		chem_effects[effect] = magnitude
 
+/mob/living/carbon/proc/remove_chemical_effect(var/effect, var/magnitude)
+	if(effect in chem_effects && magnitude)
+		chem_effects[effect] = max(0,chem_effects[effect]-magnitude)
+	else if(effect in chem_effects)
+		chem_effects[effect] = 0
```

Comment:
```suggestion
	if(effect in chem_effects)
		chem_effects[effect] = magnitude ? max(0,chem_effects[effect]-magnitude) : 0
```
Note also that a negative magnitude will instead worsen the effect. Probably fine, I think the magnitude bit would make it weird to try to mix with the proc to add effects.